### ISSUE TITLE
Using Sharppy Widgets in Seperate PySide Application

### ISSUE THREAD

**Issue by aarande** [sharppy/SHARPpy#77]
Hi guys,

I am writing a monitor program in PySide for local use in AWIPS and using the sharptab portion to calculate variables. What I am wondering is it is possible to reuse some of the widgets (i.e. hodo) in sharppy in my application. I have tried to do this by creating a profile collection and instantiating the plotHodo class and adding the widget to my layout but I keep getting a lot of paint errors as shown below. Any ideas on how I might accomplish this?
```
QPainter::begin: Paint device returned engine == 0, type: 2
QPainter::setRenderHint: Painter must be active to set rendering hints
QPainter::setRenderHint: Painter must be active to set rendering hints
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::drawPath: Painter not active
QPainter::setPen: Painter not active
QPainter::setPen: Painter not active
QPainter::drawRects: Painter not active
QPainter::drawRects: Painter not active
QPainter::setPen: Painter not active
QPainter::setFont: Painter not active
QPainter::setPen: Painter not active
QPainter::setPen: Painter not active
QPainter::drawRects: Painter not active
QPainter::drawRects: Painter not active
QPainter::setPen: Painter not active
QPainter::setFont: Painter not active
QPainter::setPen: Painter not active
QPainter::drawRects: Painter not active
QPainter::setPen: Painter not active
QPainter::drawRects: Painter not active
QPainter::setPen: Painter not active
QPainter::setFont: Painter not active
QPainter::end: Painter not active, aborted
```

**Comment by keltonhalbert** [comment]
IT should be pretty straightforward to stick the hodograph into a separate PySide Application - I'm not sure what the issue with the QPainter is. Let me see if I can dig up an old script where I display only a single window.

**Comment by aarande** [comment]
Hi Kelton...have you had a chance to look around and see if you had an example of this yet? Thanks!

**Comment by keltonhalbert** [comment]
Hi @aarande , here's some example code.

It'll yell at you in the terminal output since there's no profile collection/object present, but it gets the idea across.

With Qt, you have to have a QMainWindow - that's where the QPainter comes from, to the best of my knowledge.

```
import sys
from PySide.QtCore import *
from PySide.QtGui import *
import sharppy.viz as viz

app = QApplication(sys.argv)
mainWindow = QMainWindow()
hodo = viz.plotHodo()
mainWindow.setCentralWidget(hodo)
mainWindow.show()
sys.exit(app.exec_())
```

**Comment by aarande** [comment]
Okay I tried doing your method but I am getting an attribute error and not sure why. I have the sharppy code as a top level folder in my application source code and not installed by the way.

Traceback (most recent call last):
  File "/home/awips/PycharmProjects/Nsea/bin/StormMonitor.py", line 2, in <module>
    from widgets.stationwidget import StationWidget
  File "/home/awips/PycharmProjects/Nsea/bin/widgets/stationwidget.py", line 6, in <module>
    import sharppy.viz as viz
  File "/home/awips/PycharmProjects/Nsea/bin/sharppy/viz/__init__.py", line 1, in <module>
    from skew import plotSkewT
  File "/home/awips/PycharmProjects/Nsea/bin/sharppy/viz/skew.py", line 317, in <module>
    class plotSkewT(backgroundSkewT):
  File "/home/awips/PycharmProjects/Nsea/bin/sharppy/viz/skew.py", line 319, in plotSkewT
    parcel = Signal(tab.params.Parcel)
AttributeError: 'module' object has no attribute 'params'

Any ideas?

**Comment by aarande** [comment]
Hi Kelton...thanks for the code. This is what I had tried before but was getting errors. I updated my sharppy to the latest master branch and now it is all working. Guess there was something messing up in my older copy (Xenia release). 

I know have widgets inside of my other application and it really is awesome! Thanks for all your work on this! I heard someone let you know what I was working on (NSEA Application in AWIPS). Maybe offline I can send you some screenshots and let you know how I am using SHARPpy.
